aziotctl/src/config/mod.rs

```diff
@@ -0,0 +1,126 @@
+
+/// The ID used for the private key and cert that is used as the local CA.
+const LOCAL_CA: &str = "local-ca";
+
+/// The ID used for the private key and cert that is used as the client cert to authenticate with the EST server.
+const EST_ID_ID: &str = "est-id";
+
+/// The ID used for the private key and cert that is used as the client cert to authenticate with the EST server for the initial bootstrap.
+const EST_BOOTSTRAP_ID: &str = "est-bootstrap-id";
+
+/// The ID used for the CA cert that is used to validate the EST server's server cert.
+const EST_SERVER_CA_ID: &str = "est-server-ca";
+
+fn create_dir_all(
+    path: &(impl AsRef<std::path::Path> + ?Sized),
+    user: &nix::unistd::User,
+    mode: u32,
+) -> anyhow::Result<()> {
+    let path = path.as_ref();
+    let path_displayable = path.display();
+
+    let () = std::fs::create_dir_all(path)
+        .with_context(|| format!("could not create {} directory", path_displayable))?;
+    let () = nix::unistd::chown(path, Some(user.uid), Some(user.gid))
+        .with_context(|| format!("could not set ownership on {} directory", path_displayable))?;
+    let () = std::fs::set_permissions(path, std::os::unix::fs::PermissionsExt::from_mode(mode))
+        .with_context(|| {
+            format!(
+                "could not set permissions on {} directory",
+                path_displayable
+            )
+        })?;
+
+    Ok(())
+}
+
+fn write_file(
+    path: &(impl AsRef<std::path::Path> + ?Sized),
+    content: &[u8],
+    user: &nix::unistd::User,
+    mode: u32,
+) -> anyhow::Result<()> {
+    let path = path.as_ref();
+    let path_displayable = path.display();
+
+    let () = std::fs::write(path, content)
+        .with_context(|| format!("could not create {}", path_displayable))?;
+    let () = nix::unistd::chown(path, Some(user.uid), Some(user.gid))
+        .with_context(|| format!("could not set ownership on {}", path_displayable))?;
+    let () = std::fs::set_permissions(path, std::os::unix::fs::PermissionsExt::from_mode(mode))
+        .with_context(|| format!("could not set permissions on {}", path_displayable))?;
+
+    Ok(())
+}
+
+fn parse_manual_connection_string(
+    connection_string: &str,
+) -> Result<(String, String, Vec<u8>), String> {
+    const HOSTNAME_KEY: &str = "HostName";
+    const DEVICEID_KEY: &str = "DeviceId";
+    const SHAREDACCESSKEY_KEY: &str = "SharedAccessKey";
+
+    let mut iothub_hostname = None;
+    let mut device_id = None;
+    let mut symmetric_key = None;
+
+    for sections in connection_string.split(';') {
+        let mut parts = sections.split('=');
+        match parts
+            .next()
+            .expect("str::split always returns at least one part")
+        {
+            HOSTNAME_KEY => iothub_hostname = parts.next(),
+            DEVICEID_KEY => device_id = parts.next(),
+            SHAREDACCESSKEY_KEY => symmetric_key = parts.next(),
+            _ => (), // Ignore extraneous component in the connection string
```

Comment:
Is this the right behavior here? Shouldn't we at-least trigger some warning?

Reply:
Yes, it's the right behavior. It's architected like a query string and sometimes has other parameters for client SDKs that we don't care about.